src/main/java/org/fcrepo/auth/webac/WebACAuthorizationDelegate.java

```diff
@@ -37,10 +49,45 @@
      */
     private static final Logger LOGGER = LoggerFactory.getLogger(WebACAuthorizationDelegate.class);
 
+    private static final Map<String, URI> actionMap;
+
+    static {
+        Map<String, URI> map = new HashMap<>();
+        map.put("GET", WEBAC_MODE_READ);
+        map.put("POST", WEBAC_MODE_APPEND);
+        map.put("PUT", WEBAC_MODE_WRITE);
+        map.put("DELETE", WEBAC_MODE_WRITE);
+        map.put("PATCH", WEBAC_MODE_WRITE);
+        map.put("OPTIONS", WEBAC_MODE_READ);
+        actionMap = Collections.unmodifiableMap(map);
+    }
+
     @Override
     public boolean rolesHavePermission(final Session userSession, final String absPath,
             final String[] actions, final Set<String> roles) {
-        final boolean permit = false;
+
+        // This is not correct -- we should get it from the container or header, etc
+        final String agent = userSession.getUserID();
```

Comment:
This method could be made `protected`:
https://github.com/fcrepo4/fcrepo-module-auth-rbacl/blob/master/fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AbstractRolesAuthorizationDelegate.java#L141